# 6주차 정답 Notebook — 공정별·설비별 데이터 비교

## 1단계. Pandas 불러오고 데이터 읽기

In [1]:
import pandas as pd

df = pd.read_csv("../../data/weekly/week06/week06_equipment_comparison.csv")
df.head()

,로트번호,설비번호,공정명,작업조,온도_섭씨,압력_Pa,진공도_mTorr,처리시간_sec,합격여부
0,LOT-0722,EQ-04,산화,C조,NaN,1003.7,5.63,117.2,1
1,LOT-0331,EQ-01,증착,B조,298.1,1010.0,5.28,116.5,1
2,LOT-0106,EQ-01,식각,C조,NaN,1003.7,4.69,123.8,1
3,LOT-0656,EQ-02,산화,A조,300.1,1013.7,5.05,117.5,1
4,LOT-0778,EQ-04,포토,B조,300.5,1007.6,5.05,128.9,-1


## 2단계. 데이터 크기와 결측값 확인하기

In [2]:
print(df.shape)
print(df.isna().sum())

(240, 9)
로트번호          0
설비번호          0
공정명           0
작업조           0
온도_섭씨        10
압력_Pa         0
진공도_mTorr     0
처리시간_sec      0
합격여부          0
dtype: int64


## 3단계. 설비별 평균 온도 구하기

In [3]:
df.groupby("설비번호")["온도_섭씨"].mean()

설비번호
EQ-01    299.784746
EQ-02    299.083929
EQ-03    299.747170
EQ-04    301.143548
Name: 온도_섭씨, dtype: float64

## 4단계. 설비별 온도 표준편차 구하기

In [4]:
df.groupby("설비번호")["온도_섭씨"].std()

설비번호
EQ-01    2.394737
EQ-02    7.162701
EQ-03    2.672841
EQ-04    4.116525
Name: 온도_섭씨, dtype: float64

**질문 답**: 평균이 가장 낮은 설비는 EQ-02(299.08℃)이고, 표준편차가 가장 큰 설비도 EQ-02(7.16)다. 즉 평균은 낮아 정상적으로 보이지만, 실제로는 값이 가장 넓게 흔들리는 불안정한 설비다.

## 5단계. 설비별 평균 압력 & 공정별 평균 처리시간

In [5]:
df.groupby("설비번호")["압력_Pa"].mean()

설비번호
EQ-01    1009.175806
EQ-02    1009.373684
EQ-03    1027.391228
EQ-04    1012.273438
Name: 압력_Pa, dtype: float64

In [6]:
df.groupby("공정명")["처리시간_sec"].mean()

공정명
산화    118.797872
세정    120.291071
식각    119.109804
증착    120.072340
포토    119.966667
Name: 처리시간_sec, dtype: float64

## 6단계. 설비별·작업조별 불량률 구하기

In [7]:
fail_rate_by_eq = (df["합격여부"] == -1).groupby(df["설비번호"]).mean() * 100
fail_rate_by_eq.round(1)

설비번호
EQ-01     6.5
EQ-02    10.5
EQ-03     7.0
EQ-04    21.9
Name: 합격여부, dtype: float64

In [8]:
fail_rate_by_shift = (df["합격여부"] == -1).groupby(df["작업조"]).mean() * 100
fail_rate_by_shift.round(1)

작업조
A조     8.1
B조    17.9
C조     9.2
Name: 합격여부, dtype: float64

## 7단계. 가장 불안정한 설비 찾기

In [9]:
df.groupby("설비번호")["온도_섭씨"].std().idxmax()

'EQ-02'

## 8단계. 오늘의 분석을 한 문장으로 정리하기

> 불량률이 가장 높은 설비는 EQ-04(약 21.9%)이고, 가장 불안정한(표준편차가 가장 큰) 설비는 EQ-02(표준편차 약 7.16)이다. 두 설비는 서로 다르다 — 즉 '불량이 잦은 것'과 '값이 들쭉날쭉한 것'은 서로 다른 문제이며, 각각 따로 확인해야 한다.